plat/common/include/arm/smccc.h

```diff
@@ -0,0 +1,274 @@
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. Neither the name of the copyright holder nor the names of its
+ *    contributors may be used to endorse or promote products derived from
+ *    this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef __UKARCH_SMCCC_H__
+#define __UKARCH_SMCCC_H__
+
+#include <uk/config.h>
+
+/**
+ * Implements SMCCC up to v1.3 (ARM DEN 0028D)
+ */
+
+#define SMCCC_VERSION_1_0	0x10000U
+#define SMCCC_VERSION_1_1	0x10001U
+#define SMCCC_VERSION_1_2	0x10002U
+#define SMCCC_VERSION_1_3	0x10003U
+#define SMCCC_VERSION_MAX	SMCCC_VERSION_1_3
+
+#define SMCCC_VERSION_MAJOR_MASK  0x7fffU
+#define SMCCC_VERSION_MAJOR_SHIFT 16U
+#define SMCCC_VERSION_MINOR_MASK  0xffffU
+#define SMCCC_VERSION_MINOR_SHIFT 0U
+
+#define	SMCCC_CONDUIT_SMC	0U
+#define	SMCCC_CONDUIT_HVC	1U
+
+#define SMCCC_TYPE_YIELDING	0U
+#define SMCCC_TYPE_FAST		1U
+
+#define SMCCC_CC_32		0U
+#define SMCCC_CC_64		1U
+
+#define SMCCC_TYPE_MASK		0x80000000U
+#define SMCCC_TYPE_SHIFT	31U
+#define SMCCC_CALL_CONV_MASK	0x40000000U
+#define SMCCC_CALL_CONV_SHIFT	30U
+#define SMCCC_SERVICE_MASK	0x3f000000U
+#define SMCCC_SERVICE_SHIFT	24U
+#define SMCCC_SVE_STATE_MASK	0x10000U
+#define SMCCC_SVE_STATE_SHIFT	16U
+#define SMCCC_FNUM_MASK		0x0000ffffU
+#define SMCCC_FNUM_SHIFT	0U
+
+#define SMCCC_SVE_STATE_ON	0U
+#define SMCCC_SVE_STATE_OFF	1U
+
+#define SMCCC_FID(_type, _cc, _svc, _fnum)				\
+	((((_type) & SMCCC_TYPE_MASK) << SMCCC_TYPE_SHIFT) |		\
+	 (((_cc) & SMCCC_CALL_CONV_MASK) << SMCCC_CALL_CONV_SHIFT) |	\
+	 (((_svc) & SMCCC_SERVICE_MASK) << SMCCC_SERVICE_SHIFT) |       \
+	 ((_fnum) & SMCCC_FNUM_MASK) << SMCCC_FNUM_SHIFT)
```

Comment:
Please change the spaces in line 78 at the end to a tab 🔥 😄 

The macro is broken. The masks will kill the arguments as the macro receives unshifted values. It should be:
```suggestion
#define SMCCC_FID(_type, _cc, _svc, _fnum)				\
	((((_type) << SMCCC_TYPE_SHIFT) & SMCCC_TYPE_MASK) |		\
	 (((_cc) << SMCCC_CALL_CONV_SHIFT) & SMCCC_CALL_CONV_MASK) |	\
	 (((_svc) << SMCCC_SERVICE_SHIFT) & SMCCC_SERVICE_MASK) |	\
	 ((_fnum) << SMCCC_FNUM_SHIFT) & SMCCC_FNUM_MASK)
```

Reply:
Ooops you're right, thanks for that